packages/plugins/bundler-report/src/index.ts

```diff
@@ -2,39 +2,27 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2019-Present Datadog, Inc.
 
+import {
+    getAbsolutePath,
+    getNearestCommonDirectory,
+    getHighestPackageJsonDir,
+} from '@dd/core/helpers';
 import type { GlobalContext, PluginOptions } from '@dd/core/types';
 import path from 'path';
 
 export const PLUGIN_NAME = 'datadog-bundler-report-plugin';
 
-// From a list of path, return the nearest common directory.
-const getNearestCommonDirectory = (dirs: string[], cwd: string) => {
-    const splitPaths = dirs.map((dir) => {
-        const absolutePath = path.isAbsolute(dir) ? dir : path.resolve(cwd, dir);
-        return absolutePath.split(path.sep);
-    });
-
-    // Use the shortest length for faster results.
-    const minLength = Math.min(...splitPaths.map((parts) => parts.length));
-    const commonParts = [];
-
-    for (let i = 0; i < minLength; i++) {
-        // We use the first path as our basis.
-        const component = splitPaths[0][i];
-        if (splitPaths.every((parts) => parts[i] === component)) {
-            commonParts.push(component);
-        } else {
-            break;
-        }
+// Compute the CWD based on a list of directories and the outDir.
```

Comment:
cf: https://github.com/DataDog/build-plugins/pull/145/files#r1991128004

For a more detailed explanation.